src/app/dao/tests/PartDAO.spec.ts

```diff
@@ -60,43 +74,249 @@ describe('PartDAO', () => {
             // When checking if the user has an active part
-            const result: boolean = await dao.userHasActivePart(username);
+            const result: boolean = await partDAO.userHasActivePart(UserMocks.OPPONENT_MINIMAL_USER);
             // Then it should return true
             expect(result).toBeTrue();
         });
         it('should return true when user has an active part as player one', async() => {
             // Given a part where user is player zero
-            await dao.create({ ...part, playerOne: username });
+            await partDAO.create({ ...part, playerOne: UserMocks.OPPONENT_MINIMAL_USER });
             // When checking if the user has an active part
-            const result: boolean = await dao.userHasActivePart(username);
+            const result: boolean = await partDAO.userHasActivePart(UserMocks.OPPONENT_MINIMAL_USER);
             // Then it should return true
             expect(result).toBeTrue();
         });
         it('should return false when the user has no active part', async() => {
             // Given a part where the user is not active
-            await dao.create(part);
+            await partDAO.create(part);
             // When checking if the user has an active part
-            const result: boolean = await dao.userHasActivePart(username);
+            const result: boolean = await partDAO.userHasActivePart(UserMocks.OPPONENT_MINIMAL_USER);
             // Then it should return false
             expect(result).toBeFalse();
 
         });
         afterEach(async() => {
-            await signOut(TestBed.inject(Auth));
+            await signOut();
         });
     });
+    it('should allow a verified user to create a part', async() => {
+        // Given a verified user
+        await createConnectedGoogleUser('foo@bar.com', 'creator');
+        // When creating a part
+        const result: Promise<string> = partDAO.create(PartMocks.INITIAL);
+        // Then it should succeed
+        await expectAsync(result).toBeResolved();
+
+    });
+    it('should forbid a non-verified user to create a part', async() => {
+        // Given a non-verified user
+        const token: string = '{"sub": "bar@bar.com", "email": "bar@bar.com", "email_verified": false}';
+        const credential: FireAuth.UserCredential =
+            await FireAuth.signInWithCredential(TestBed.inject(FireAuth.Auth),
+                                                FireAuth.GoogleAuthProvider.credential(token));
+        await userDAO.set(credential.user.uid, { verified: false, username: 'user' });
+
+        // When creating a part
+        const result: Promise<string> = partDAO.create(PartMocks.INITIAL);
+        // Then it should fail
+        await expectFirebasePermissionDenied(result);
+    });
+    it('should forbid player to change playerZero/playerOne/beginning once a part has started', async() => {
+        // Given a part that has started (i.e., turn >= 0), and a player (here creator)
```

Comment:
I would say it's started when there is a beginningTime, we could in the future prefer part NOT to be at turn -1 anymoment